build_tools/github_actions/runner/README.md

```diff
@@ -0,0 +1,186 @@
+scoped [de]registration token. We
+https://github.com/google-github-actions/github-runner-token-proxy for the
+proxy. You can see its docs for more details.
+
+## Service Accounts
+
+The presubmit and postsubmit runners run as different service accounts depending
+on their trust level. Presubmit runners are "minimal" trust and postsubmit
+runners are "basic" trust, so they run as
+`github-runner-minimal-trust@iree-oss.iam.gserviceaccount.com` and
+`github-runner-basic-trust@iree-oss.iam.gserviceaccount.com`, respectively.
+
+## Passing Artifacts
+
+Using GitHub's [artifact actions](https://github.com/actions/upload-artifact)
+with runners on GCE turns out to be prohibitively slow (see discussion in
+https://github.com/iree-org/iree/issues/9881). Instead we use our own
+[Google Cloud Storage](https://cloud.google.com/storage) (GCS) buckets to save
+artifacts from jobs and fetch them in subsequent jobs:
+`iree-github-actions-presubmit-artifacts` and
+`iree-github-actions-postsubmit-artifacts`. Each runner group's service account
+has acces only to the bucket for its group. Artifacts are indexed by the
+workflow run id and attempt number, so that they do not collide. Subsequent jobs
+should *not* make assumptions about where an artifact was stored however,
+instead querying the outputs of the job that created it (which should always
+provide such an output). This is both to promote DRY principles and for subtle
+reasons like a rerun of a failed job may be on run attempt 2, but fetching
+artifacts from a job dependency that succeeded on attempt 1 and therefore did
+not rerun and recreate the artifacts indexed by the new attempt.
+
+## Labels
+
+The GitHub Actions Runners are identified with
+[labels](https://docs.github.com/en/enterprise-cloud@latest/actions/hosting-your-own-runners/using-labels-with-self-hosted-runners)
+that indicate properties of the runner. Some of the labels are automatically
+generated from information about the runner on startup, such as its GCP zone and
+hostname, others match GitHub's standard labels, like the OS, and some are
+injected as custom labels via metadata, like whether the VM is optimized for CPU
+or GPU usage. All self-hosted runners receive the `self-hosted` label.
+
+Note that when setting where a job runs, any runner that has all the specified
+labels can pick up a job. So if you leave off the runner-group, for instance,
+the job will non-deterministically try to run on presubmit or postsubmit
+runners. We do not currently have a solution for this problem other than careful
+code authorship and review.
+
+## Examining Runners
+
+The runners for iree-org can be viewed in the
+[GitHub UI](https://github.com/organizations/iree-org/settings/actions/runners).
+Unfortunately, only organization admins (currently @stellaraccident and
+@GMNGeoffrey) have access to this page. Organization admin gives very broad
+privileges, so this set is necessarily kept very small by Google security
+policy.
```

Comment:
nit: not a huge fan of including specific usernames here. I'd just leave those off for now, but we could also create a group like `iree-org-admins` if we want something to refer to by name

Reply:
Eone